sig-cli/charter.md

```diff
@@ -5,25 +5,81 @@ the Roles and Organization Management outlined in [sig-governance].
 
 ## Scope
 
-The Command Line Interface SIG (SIG CLI) is responsible for kubectl and
-related tools. This group focuses on general purpose command line tools and
-libraries to interface with Kubernetes API's.
-
 ### In scope
 
-SIG CLI [README]
-
-#### Code, Binaries and Services
-
-SIG CLI code include general purpose command line tools and binaries for working
-with Kubernetes API's. Examples of these binaries include: [kubectl and kustomize].
+The Command Line Interface SIG (SIG CLI) is responsible for kubectl and
+related tools that are developed by the SIG.  See kubectl
+[design principles](design-principles.md) for the focus of kubectl functionality.
+
+This group focuses on command line tooling for working with Kubernetes APIs and Resource
+Config.  This includes both generalized tooling for working with Resources, Resource Config
+and Resource Types (e.g. using resource / object metadata, duck-typing, openapi, discovery, 
+scale and status subresources, etc) as well as tooling for working with specific Kubernetes
+APIs (e.g. `logs`, `exec`, `create configmap`).
+
+The scope includes both low level tooling that may be used by scripts or higher-level
+solutions, as well as higher level porcelain to reduce user friction for simple,
+common, difficult or important tasks performed by users (e.g. `edit`, `top`, `wait`).  The
+scope also includes publishing a subset the libraries which were used to develop the tooling
+itself.
+
+The decision whether to publish specific functionality as part of kubectl,
```

Comment:
I don't think that this is true, I think that SIG-Architecture should be involved in project cross-cutting decisions like this.

Reply:
Over a year ago there, at the guidance of the steering committee, there was a shift for some opinions (e.g., Helm and Kompose) to be treated as ecosystem rather than a core part of the project. This allowed us to say one way was not the way and encourage competition. This is walking close to those opinion spaces depending on the solutions created.

Would that make it something SIG Arch would be interested in or provide further guidance for SIG CLI (or SIG Apps scope)?

Reply:
Another example: kubeadm, which is in releases, and is also adding commands.

Reply:
Went ahead and created a PR for the scope document:
https://github.com/kubernetes/community/pull/3180

Reply:
@bgrant0607 The plan, as I understand it, is to move kubectl out of kubernetes/kubernetes and have its own release schedule. I think this is a good thing and like that plan. Once that happens, won't it fall out of "core"?

My context for asking about this is, it's already possible for me to do all of my interactions with a cluster without directly interfacing with kubectl and, given some more time, it will be possible to do so without it in the loop at all (even being called as a background tool). It may already be possible but not in the toolchain I personally use.

Reply:
> The plan, as I understand it, is to move kubectl out of kubernetes/kubernetes and have its own release schedule.

That is correct, also we're pushing hard to move majority of functionality over to the server, which means all clients will have equal chances to interact with k8s API. The fact that kubectl will be reference implementation does not change, though.

Reply:
I agree with Brian that CLI surface area changes should require API review.  The CLI surface area effective _is_ an API (I think Brian made this point in a meeting I was in a long, long time ago)

And, much like APIs, the kubectl tool has significant impact across the entire community, I think that clearly makes it in the territory of SIG-Architecture.

Reply:
I did not read that in Brian's comment, I've seen it more like a questions than a statement. A questions that has not been answered.

Reply:
In the charter it notes:

> The scope includes both low level tooling that may be used by scripts ...

If scripts are meant to use kubectl (and they do today) doesn't that make the commands, flags, and other arguments an API to those scripts?

Reply:
Yes, and are included in https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md#what-apis-need-to-be-reviewed

> What parts of a PR are "API changes"?
> * Configuration files, flags, and command line arguments are all part of our user and script facing APIs and must be reviewed.

Reply:
And covered by deprecation.

https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli

Reply:
That's what the sig is following on a daily basis.